src/toil/spark/spark_cluster.py

```diff
@@ -0,0 +1,409 @@
+
+        Job.Service.__init__(self, memory = self.memory, cores = self.cores)
+
+
+    def start(self, fileStore):
+        """
+        Start spark and hdfs master containers
+
+        :param fileStore: Unused
+        """
+        
+        _log.info("Started Spark master container.")
+        self.sparkContainerID = _docker_call(no_rm = True,
+                                             work_dir = os.getcwd(),
+                                             tool = "quay.io/ucsc_cgl/apache-spark-master:1.5.2",
+                                             docker_parameters = ["--net=host",
+                                                                  "-d",
+                                                                  "-v", "/mnt/ephemeral/:/ephemeral/:rw",
+                                                                  "-e", "SPARK_MASTER_IP="+self.hostname,
+                                                                  "-e", "SPARK_LOCAL_DIRS=/ephemeral/spark/local",
+                                                                  "-e", "SPARK_WORKER_DIR=/ephemeral/spark/work"],
+                                             tool_parameters = [self.hostname],
+                                             sudo = self.sudo,
+                                             check_output = True)[:-1]
+        _log.info("Started HDFS Datanode.")
+        self.hdfsContainerID = _docker_call(no_rm = True,
+                                            work_dir = os.getcwd(),
+                                            tool = "quay.io/ucsc_cgl/apache-hadoop-master:2.6.2",
+                                            docker_parameters = ["--net=host",
+                                                                 "-d"],
+                                            tool_parameters = [self.hostname],
+                                            sudo = self.sudo,
+                                            check_output = True)[:-1]
+
+        return self.hostname
+
+
+    def stop(self, fileStore):
+        """
+        Stop and remove spark and hdfs master containers
+
+        fileStore: Unused
+        """
+        
+        sudo = []
+        if self.sudo:
+            sudo = ["sudo"]
+            
+        subprocess.call(sudo + ["docker", "exec", self.sparkContainerID, "rm", "-r", "/ephemeral/spark"])
+        subprocess.call(sudo + ["docker", "stop", self.sparkContainerID])
+        subprocess.call(sudo + ["docker", "rm", self.sparkContainerID])
+        _log.info("Stopped Spark master.")
+
+        subprocess.call(sudo + ["docker", "stop", self.hdfsContainerID])
+        subprocess.call(sudo + ["docker", "rm", self.hdfsContainerID])
+        _log.info("Stopped HDFS namenode.")
+
+        return
+
+
+    def check(self):
+        """
+        Checks to see if Spark master and HDFS namenode are still running.
+        """
+        
+        _checkContainerStatus(self.sparkContainerID, self.hdfsContainerID)
```

Comment:
Prefer to include "return _checkContainerStatus(self.sparkContainerID, self.hdfsContainerID)"